scipy/stats/_discrete_distns.py

```diff
@@ -320,9 +325,11 @@ class geom_gen(rv_discrete):
 
         f(k) = (1-p)^{k-1} p
 
-    for :math:`k \ge 1`.
+    for :math:`k \ge 1`, :math:'0 \leq p \leq 1'
```

Comment:
Strict inequality on the low end:
```suggestion
    for :math:`k \ge 1`, :math:`0 < p \leq 1`
```

Reply:
This also has the same `_argcheck` as negative binomial: 
```python 
    def _argcheck(self, p):
        return (p <= 1) & (p >= 0)
`